web/src/containers/ActivityDetailExpenses.js

```diff
@@ -0,0 +1,156 @@
+                <th className="col-4">Description</th>
+                {years.map(year => (
+                  <th key={year} className="col-2">
+                    {year} Cost
+                  </th>
+                ))}
+                <th className="col-1" />
+              </tr>
+            </thead>
+            <tbody>
+              {expenses.map((expense, i) => (
+                <tr key={expense.idx}>
+                  <td className="mono">{i + 1}.</td>
+                  <td>
+                    <select
+                      className="m0 select"
+                      value={expense.category}
+                      onChange={this.handleChange(i, 'category')}
+                    >
+                      <option>Expense A</option>
+                      <option>Expense B</option>
+                      <option>Expense C</option>
+                      <option>Other</option>
+                    </select>
+                  </td>
+                  <td>
+                    <Textarea
+                      id={`expense-${i}-desc`}
+                      name={`expense-${i}-desc`}
+                      label="Describe the expense"
+                      rows="3"
+                      spellCheck="true"
+                      value={expense.desc}
+                      onChange={this.handleChange(i, 'desc')}
+                    />
+                  </td>
+                  {years.map(year => (
+                    <td key={year}>
+                      <Input
+                        name={`expense-${i}-${year}-cost`}
+                        label={`Cost for ${year}`}
+                        type="number"
+                        value={expense.years[year]}
+                        onChange={this.handleYearChange(i, year)}
+                      />
+                    </td>
+                  ))}
+                  <td className="center align-middle">
+                    <button
+                      type="button"
+                      className="btn btn-outline border-silver px1 py-tiny"
+                      title="Remove Expense"
+                      onClick={() => removeExpense(activityID, i)}
+                    >
+                      ✗
+                    </button>
+                  </td>
+                </tr>
+              ))}
+            </tbody>
+          </table>
+        </div>
+        <button
+          type="button"
+          className="btn btn-primary bg-black"
+          onClick={() => addExpense(activityID)}
+        >
+          Add expense
+        </button>
+      </Collapsible>
+    );
+  }
+}
+
+ActivityDetailExpenses.propTypes = {
+  activity: PropTypes.object.isRequired,
+  years: PropTypes.array.isRequired,
+  addExpense: PropTypes.func.isRequired,
+  removeExpense: PropTypes.func.isRequired,
+  updateActivity: PropTypes.func.isRequired
+};
+
+const mapStateToProps = ({ activities: { byId } }, { aId }) => {
+  const { expenses } = byId[aId];
+  const expenseYears = expenses
```

Comment:
this is fine for now; i think soon we want years to be an array field keyed to the *apd* that applies to all of their children activities, but this will be easy to sub in when we get there